vcd/resource_vcd_ip_space.go

```diff
@@ -0,0 +1,427 @@
+
+func resourceVcdIpSpaceCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+	log.Printf("[TRACE] IP Space creation initiated")
+
+	ipSpaceConfig, err := getIpSpaceType(d)
+	if err != nil {
+		return diag.Errorf("could not get IP Space type: %s", err)
+	}
+
+	createdIpSpace, err := vcdClient.CreateIpSpace(ipSpaceConfig)
+	if err != nil {
+		return diag.Errorf("error creating IP Space: %s", err)
+	}
+
+	d.SetId(createdIpSpace.IpSpace.ID)
+
+	return resourceVcdIpSpaceRead(ctx, d, meta)
+}
+
+func resourceVcdIpSpaceUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+	log.Printf("[TRACE] IP Space update initiated")
+
+	ipSpaceConfig, err := getIpSpaceType(d)
+	if err != nil {
+		return diag.Errorf("could not get IP Space type: %s", err)
+	}
+
+	ipSpace, err := vcdClient.GetIpSpaceById(d.Id())
+	if err != nil {
+		return diag.Errorf("error finding IP Space by ID '%s': %s", d.Id(), err)
+	}
+
+	ipSpaceConfig.ID = d.Id()
+	_, err = ipSpace.Update(ipSpaceConfig)
+	if err != nil {
+		return diag.Errorf("error updating IP Space: %s", err)
+	}
+
+	return resourceVcdIpSpaceRead(ctx, d, meta)
+}
+
+func resourceVcdIpSpaceRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+	log.Printf("[TRACE] IP Space read initiated")
+
+	ipSpace, err := vcdClient.GetIpSpaceById(d.Id())
+	if err != nil {
+		if govcd.ContainsNotFound(err) {
+			d.SetId("")
+			return nil
+		}
+		return diag.Errorf("error finding IP Space by ID '%s': %s", d.Id(), err)
+	}
+
+	err = setIpSpaceData(d, ipSpace.IpSpace)
+	if err != nil {
+		return diag.Errorf("error storing IP Space state: %s", err)
+	}
+
+	return nil
+}
+
+func resourceVcdIpSpaceDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+	log.Printf("[TRACE] IP Space deletion initiated")
+
+	ipSpace, err := vcdClient.GetIpSpaceById(d.Id())
+	if err != nil {
+		return diag.Errorf("error finding IP Space by ID '%s': %s", d.Id(), err)
+	}
+
+	err = ipSpace.Delete()
+	if err != nil {
+		return diag.Errorf("error deleting IP space by ID '%s': %s", d.Id(), err)
+	}
+
+	return nil
+}
+
+// resourceVcdIpSpaceImport has two cases:
+// * Import global (provider level) IP Space - just a name is required
+// * Import Private IP space for an Organization - org-name.ip-space-name is required
+func resourceVcdIpSpaceImport(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	log.Printf("[TRACE] IP Space import initiated")
+
+	resourceURI := strings.Split(d.Id(), ImportSeparator)
+	if len(resourceURI) != 1 && len(resourceURI) != 2 {
+		return nil, fmt.Errorf("resource name must be specified as ip-space-name or org-name.ip-space-name")
+	}
+	vcdClient := meta.(*VCDClient)
+
+	var ipSpace *govcd.IpSpace
+
+	if len(resourceURI) == 2 {
```

Comment:
Could you add a comment describing this if statement? Also, maybe you could decouple them into two so it's more clear that those are two distinct cases of import

Reply:
Changed